src/shogun/mathematics/graph/test/GraphTest.h

```diff
@@ -54,8 +53,13 @@ class GraphTest : public ::testing::Test
 	std::set<GRAPH_BACKEND> m_backends;
 };
 
+
 using GraphTypes = ::testing::Types<
-    bool, int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t,
-    uint64_t, float32_t, float64_t>;
+    shogun::graph::BooleanType, shogun::graph::UInt8Type,
+	shogun::graph::Int8Type, shogun::graph::UInt16Type,
+	shogun::graph::Int16Type, shogun::graph::UInt32Type,
+	shogun::graph::Int32Type, shogun::graph::Int64Type,
+	shogun::graph::UInt64Type, shogun::graph::Float32Type,
+	shogun::graph::Float64Type>;
```

Comment:
Somewhat unrelated, but I am thinking if it would be good to add ::testing::Types, so we can exclude bool, or just have bool, and not unsigned, and so on..? Most tests have a `if constexpr` which doesn't test anything in one branch...

Reply:
imo this is doable, with a bit different gtest construct, i.e. test suites.